src/reformat/coeffs.rs

```diff
@@ -32,6 +32,35 @@ pub fn calculate_yuv_coefficients(
     color_primaries: ColorPrimaries,
     matrix_coefficients: MatrixCoefficients,
 ) -> [f32; 3] {
-    calculate_yuv_coefficients_from_cicp(color_primaries, matrix_coefficients)
-        .unwrap_or([0.299, 0.114, 0.587])
+    // Return known coefficients or fall back to BT.601.
+    calculate_yuv_coefficients_from_cicp(color_primaries, matrix_coefficients).unwrap_or(
+        calculate_yuv_coefficients_from_cicp(color_primaries, MatrixCoefficients::Bt601).unwrap(),
+    )
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn yuv_coefficients() {
+        fn assert_near(a: [f32; 3], b: [f32; 3]) {
```

Comment:
we might need this in more places as we add more functions and tests.

maybe use a crate like this: https://docs.rs/assert_float_eq/latest/assert_float_eq/ ?

Reply:
Sure, once we need it in multiple places. Bringing a crate to save 40 chars in a single line is not worth it yet I think.

Reply:
This will be a dev-only dependency since it is only used for tests. So i think it's fine to bring it in.

The Rust world seems to operate more like python where you simply add dependencies (no matter how small) and have pip deal with them instead of re-implementing them. Especially if it is not a dependency of the core library, then we don't really care how small or big it is.

Reply:
I added `assert_float_eq = "1.1.3"` in `[dev-dependencies]` of `Cargo.toml`.  
I added `use assert_float_eq::assert_f32_near;` in the `mod tests` and replaced the line with `assert_f32_near!(a[i], b[i]);`.  
I get:
```rs
error: cannot find macro `afe_near_error_msg` in this scope
  --> src/reformat/coeffs.rs:50:17
   |
50 |                 assert_f32_near!(a[i], b[i]);
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `assert_f32_near` (in Nightly builds, run with -Z macro-backtrace for more info)

error: cannot find macro `afe_is_f32_near` in this scope
   --> src/reformat/coeffs.rs:50:17
    |
50  |                 assert_f32_near!(a[i], b[i]);
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: a macro with a similar name exists: `assert_f32_near`
    |
   ::: /home/yguyon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/assert_float_eq-1.1.3/src/lib.rs:626:1
    |
626 | macro_rules! assert_f32_near {
    | ---------------------------- similarly named macro `assert_f32_near` defined here
    |
    = note: this error originates in the macro `assert_f32_near` (in Nightly builds, run with -Z macro-backtrace for more info)
```
Shouldn't rust deal with bringing dependencies of dependencies itself? Or did I call the macro incorrectly? The [examples](https://docs.rs/assert_float_eq/latest/assert_float_eq/macro.assert_f32_near.html#examples) do not give extra details.

Anyway this can be added in another PR.

Reply:
You don't need to add the "use" statement for the macros since they are generated at compile time. I think it will work if you simply remove the "use" statement.

Reply:
I get
```rust
error: cannot find macro `assert_f32_near` in this scope
  --> src/reformat/coeffs.rs:49:17
   |
49 |                 assert_f32_near!(a[i], b[i]);
   |                 ^^^^^^^^^^^^^^^
   |
help: consider importing this macro
   |
43 +     use assert_float_eq::assert_f32_near;
   |
```

Reply:
Heh. Ok you can merge this PR as-is. I will look into it later.